promise/assign4.1WithPromise.js

```diff
@@ -0,0 +1,47 @@
+const fs = require('fs');
+var file1Content = ` file 1 content goes 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 
+  1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 
+  1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 `;
+
+var writeInFile1 = () => {
+  return new Promise ( (resolve,reject) => {
+    fs.writeFile("file1.txt", file1Content, (err,data) => {
+      if(err)
+        reject("error occured in reading file1");
+      resolve("file 1 written initially");
```

Comment:
if you don't use return statement in line#10 , then the execution flow would not be broken and resolve would get called!